docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+the static void method `executeFilterCommand` that will be called first when the user enters a filter command. 
+In the `executeFilterCommand` method, it will make use of the enumeration `FilterType` to get the filter type (by `WORD_TYPE`, 
+`STARTING STRING` or `INCLUDING_STRING`). After that, the method will use the `FilterCommandSlicer` static methods `isNewFilter` 
+to determine whether the user wants to continue on the last filtered list or start a new filter on an entire word bank. 
+`executeFilterCommand` will also check whether the user has entered a print limit using `FilterCommandSlicer`'s method 
+`getWordPrintLimitFromFilterCommand`. Subsequently, depending on the filter mode, `getTargetedWordTypes` or `getTargetedStringTags` will be called 
+and the returned array of strings will be passed to `WordsFilter`’s static methods `filterByType`, `filterByStartingString` and `filterByIncludedString`.
+Lastly, `filterCommandExecutor` will call the method `printFilterList`.
+
+The following sequence diagram shows how the components interact with each other for the scenario where the user issues 
+the command `filter -continue by\start limit\10 -cs -cg.`
+
+![UML Filter word sequence diagram](graphics/diagrams/Sequence_FilterWords.png)
+
+<p align = "center"><i><b>Figure 5: Interactions between components for the command filter -continue by\start limit\10 -cs -cg</b></i></p>
+   
+## Bunny class family
+![UML Bunny class diagram](graphics/diagrams/Class_diagram_bunny.png)
+<center><i>Figure 6:  Bunny ideas UML Class Diagram</i></center>
+The above class diagram describes the overall architecture of the bunny list functionalities. Recall that the term bunny refers to  plot ideas that have yet to be devloped. 
+The above classes provide the functionality of storing such ideas in an organised manner that can easily be searched, saved and loaded.
+
+The `BunnyList` class has the public ArrayList of bunnies `bunniesList` that is accessed by the `DeleteBunny` class method `deleteBunny` which removes 
+a selected bunny from the `bunniesList` ArrayList. Similarly, `bunniesList` is also accessed by the `BunnyFilter` class which 
+contains the `filterBunny` function which can filter through the list and obtain bunnies with specified keywords in the 
+idea or the genre using the command `filter bunny i\IDEA g\GENRE`, where the user may choose to omit either the `IDEA` 
+or the `GENRE` when running the command. 
+
+The `BunnySaver` class accesses the `bunniesList` and overwrites the current `bunny.txt` file in the data directory, 
+saving all `Bunny` objects into the file using the `saveAllBunny`  method. Bunny objects saved in that file can then 
+be read by the `BunnyLoader` class and added into the `bunniesList` ArrayList each time the program is started up, which is done 
+by calling the `loadBunnyFile` method.
+
+The `GenBunny` class can access the `bunniesList` as well. The function `pickRandomBunny` from the `GenBunny` class first randomly 
```

Comment:
Should be 'The method `pickRandomBunny`.....' . There are other instances of the word function being used and I think it would be good to change all of them to method.